components/Python/Scikit-learn/loan_demo/loan-infer-comp/loan-analysis-inference.py

```diff
@@ -0,0 +1,249 @@
+        else:
+            trans_feature_names.append(column_names[mod_el])
+    trans_feature_names1 = np.asarray(trans_feature_names)
+    model_FE_index = np.argsort(final_model.steps[-1][1].feature_importances_)[::-1][:num_features]
+    feat_eng = pd.DataFrame({'Name': trans_feature_names1[model_FE_index],
+                             'Importance': final_model.steps[-1][1].feature_importances_[model_FE_index]})
+    print("Feature Importance for " + str(title_name) + "\n", feat_eng)
+    # Output Feature Importance as a BarGraph using MCenter
+    export_bar_table(trans_feature_names1[model_FE_index],
+                     final_model.steps[-1][1].feature_importances_[model_FE_index],
+                     "Feature Importance for " + str(title_name))
+
+
+def get_psi(v1, v2, num1=10):
+    """
+    calculate PSI.
+
+    :param v1: vector 1
+    :param v2: vector 2
+    :param num1: number of bins
+    :return: PSI Value
+    """
+    rank1 = pd.qcut(v1, num1, labels=False, duplicates="drop") + 1
+    num = min(num1, max(rank1))
+
+    basepop1 = pd.DataFrame({'v1': v1, 'rank1': rank1})
+
+    quantiles = basepop1.groupby('rank1').agg({'min', 'max'})
+    quantiles.loc[1, 'v1'][0] = 0
+
+    currpop = pd.DataFrame({'v2': v2, 'rank1': [1] * v2.shape[0]})
+    for i in range(2, num + 1):
+        currpop.loc[currpop['v2'] >= quantiles['v1'].loc[i][0], 'rank1'] = i
+        quantiles.loc[i - 1, 'v1'][1] = quantiles.loc[i, 'v1'][0]
+    quantiles.loc[num, 'v1'][1] = 1
+
+    basepop2 = basepop1.groupby('rank1').agg({'count'})
+    basepop2 = basepop2.rename(columns={'count': 'basenum'})
+
+    currpop2 = currpop.groupby('rank1').agg({'count'})
+    currpop2 = currpop2.rename(columns={'count': 'currnum'})
+
+    nbase = basepop1.shape[0]
+    ncurr = currpop.shape[0]
+
+    mrged1 = basepop2['v1'].join(currpop2['v2'], how='left')
+    if mrged1.shape[0] > 1:
+        mrged1.loc[mrged1.currnum.isna(), "currnum"] = 0
+
+    mrged2 = mrged1.join(quantiles['v1'], how='left')
+
+    mrged3 = mrged2
+    mrged3['basepct'] = mrged3.basenum / nbase
+    mrged3['currpct'] = mrged3.currnum / ncurr
+
+    mrged4 = mrged3
+    mrged4['psi'] = (mrged4.currpct - mrged4.basepct) * np.log((mrged4.currpct / mrged4.basepct))
+
+    print("Merged DF: {}".format(mrged4))
+
+    tot_PSI = sum(mrged4.psi[mrged4.psi != float('inf')])
+    final_table = mrged4
+    return tot_PSI, final_table
+
+
+def export_bar_table(bar_names, bar_data, title_name):
+    """
+    This function provides a bar_graph for a bar type data at MCenter data scientist view
+    :param bar_names: Bar graph names
+    :param bar_data: Bar graph data.
+    :param title_name: Title of the bar Graph
+    :return:
+    """
+    bar_graph_data = BarGraph().name(title_name).cols(
+        bar_names.astype(str).tolist()).data(
+        bar_data.tolist())
+    mlops.set_stat(bar_graph_data)
+
+
+def main():
```

Comment:
Is it supposed to be a standalone component? Why not breaking it up (if possible) to multiple connectable components?
In any case, you should declare it as a standalone in the component json.

Reply:
Joy will break it to components afterwards and source from S3. this is a standalone component